### ISSUE TITLE
Add Arrow Support

### ISSUE THREAD

**Issue by sunchao** [sunchao/parquet-rs#186]
This is the umbrella ticket to track adding Apache Arrow support. Tasks:

- Adding Arrow schema converter (#185).

**Comment by sunchao** [comment]
Thanks @liurenjie1024 . Updated the description for some potential tasks.

**Comment by sadikovi** [comment]
I suggest adding an item to update the existing doc to reflect the addition of arrow reader/writer.

**Comment by andygrove** [comment]
DataFusion has code for loading parquet into arrow ... might be worth
looking at

**Comment by sunchao** [comment]
@sadikovi Thanks - added. 
@andygrove cool - will take a look.

**Comment by liurenjie1024** [comment]
@andygrove Yes, I'll take that as a reference. Also I'll also reference the cpp implementation of arrow adapter of parquet.

**Comment by andygrove** [comment]
I am very interested  in this. I am wondering if we can add a generic reader trait to the main arrow project and then have an implementation in parquet-rs.

I have a CSV reader for arrow that could be published as a separate crate and implement the same trait.
